### DISCUSSION TITLE
what is `volar.takeOverMode.extension`

### DISCUSSION THREAD

**Discussion by mesqueeb** [vuejs/language-tools#2275] [Q&A]
I saw in the recent [CHANGELOG](https://github.com/johnsoncodehk/volar/blob/master/CHANGELOG.md#1020-202214)

> feat: add volar.takeOverMode.extension setting for choice extension that takeover *.ts language support

I wonder, what does this mean exactly? What does this VSCode setting do exactly?

**Comment by johnsoncodehk** [comment]
This is a feature added for future, because there may be more extensions developed based on volar that support takeover mode(For example: https://github.com/volarjs/angular-language-tools), so a setting is required to specify which extension should enable takeover mode.

**Reply by mesqueeb** [reply]
Do we haze to enable it in Vue projects? Will it give us any benefits now?

**Reply by johnsoncodehk** [reply]
You don't need to change the default setting for vue project. (Unless you want to use https://github.com/volarjs/angular-language-tools, you need to set `"volar.takeOverMode.extension": "johnsoncodehk.vscode-angular"`.)

This setting is currently not beneficial, but some future cross-extensions cooperation functions need to be based on this setting.

For example, Volar is exploring with Astro to support both Vue and Astro in one language server： https://github.com/withastro/language-tools/issues/451
